src/MentorBot.Business/Services/CognitiveService.cs

```diff
@@ -41,6 +47,9 @@ public Task<CognitiveTextAnalysisResult> ProcessAsync(ChatEvent chatEvent)
                 text.EndsWith("?", StringComparison.InvariantCulture) ||
                 Regex.IsMatch(text, "^(([Ww]hat)|([Ww]here)|([Hh]ow)|([Ww]hy)|([Ww]ho))\\s");
 
+            var dataQuestion = QuestionFactory.GetQuestion(text);
+            _questionRepository.AddAsync(dataQuestion);
```

Comment:
I don not see where you are calling SaveChangesAsync()?

Reply:
QuestionRepository.cs - :29 AddEntityAsync(question);

Reply:
It is not a good idea to save every time when you make an add or update.
We may want to do that in batches.

Reply:
Good idea. Do you have any solution about that in mind?
I am seeing this problem to be solved like we are tracking the number queries and stack them. Once they reach the set number of database queries or the time limit then we send them for database committing.
I am not sure whether we will be able to implement such an approach with Azure functions. @rosenkolev

Reply:
You can call SaveAsync() after the AddAsync()